test/e2e/apimachinery/webhook.go

```diff
@@ -601,8 +629,11 @@ func deployWebhookAndService(f *framework.Framework, image string, context *cert
 				"--tls-private-key-file=/webhook.local.config/certificates/tls.key",
 				"--alsologtostderr",
 				"-v=4",
+				fmt.Sprintf("--port=%d", servicePort),
 			},
 			Image: image,
+			// Use same port for service and containers.
+			Ports: []v1.ContainerPort{{ContainerPort: servicePort}},
```

Comment:
it looks like we were already using a non-standard port for the service (which is good). I expected this PR to just change the container port to something that was not 443 and not the same as the service port (e.g. 9444). I would just hard code that into the pod and service spec.... I think that would make this PR a ~3 line change

Reply:
I've rewritten the PR to just swap the container port with a non-default one which is much simpler.  It's still more than 3 lines because I removed the global port variable and replaced it with a context scoped one which required threading a param through the stack, which is conceptually trivial but took quite a few lines to do.

Reply:
see latest PR update